docs/source/user-guide/vision/data-classes/index.rst

```diff
@@ -0,0 +1,56 @@
+.. _data_classes:
+
+====================
+Data Classes
+====================
+
+The different data classes in the deepchecks.vision module were designed to enable access to the data in a unified
+format, while making the formatting code easy and maintainable. They are responsible for converting the data (whether
+these are images, labels, or predictions) to a format deepchecks can work with.
+
+.. toctree::
+    :maxdepth: 2
+
+    VisionData
+    ClassificationData
+    DetectionData
+
+Why Is This Important?
+========================
+
+There is no standard when it comes to data formats in computer vision. Each model can have its own supported data format
+for the output predictions, and each new benchmark dataset can propose a new labeling format. Even the images formats
+can be different from each other (PIL, OpenCV, etc.).
+
+In order to run a model and data-agnostic comprehensive testing suites, the data should be in an accepted format the
+checks can work with. The data class objects provide a structured and repeatable way to do that, and are an important
+part of the vision module.
+
+What Do You Need to Implement?
+==============================
+
+Generally, you will need to implement a data class that inherits from one of the supported classes in the vision module,
+and implement 3 functions:
+
+    - `batch_to_images`: Returns a list of images in the correct format from a batch of data.
+    - `batch_to_labels`: Returns a list of labels in the correct format from a batch of data.
+    - `infer_on_batch`: Returns a list of a model's predictions in the correct format from a batch of data.
+
+Please see specific examples in the `VisionData <VisionData.rst>`_, `ClassificationData <ClassificationData.rst>`_,
+and `DetectionData <DetectionData.rst>`_ classes.
+
+Validating The Correctness of Your Data Class
+=============================================
+While implementing the data class, you may need a way to test your work is correct.
```

Comment:
Again here the word "correct" or "correctness" are weird (as in the title). Would discuss the format instead of a general notion of "correct"